src/common/ant.ts

```diff
@@ -93,11 +100,50 @@ export class AoANTReadable implements AoANTRead {
     }
 
     this.processId = this.process.processId;
+    this.hyperbeamUrl = config.hyperbeamUrl || 'https://permanode.xyz'; // TODO: replace this with hyperbeam.ario.permaweb.services once deployed
+    this.checkHyperBeamPromise = this.checkHyperBeamCompatibility();
+  }
+
+  private async checkHyperBeamCompatibility(): Promise<boolean> {
+    const res = await fetch(
+      `https://permanode.xyz/${this.processId}~process@1.0/now/cache`,
+      {
+        method: 'HEAD',
+      },
+    );
+
+    if (res.ok) {
+      return true;
+    }
+
+    return false;
```

Comment:
Rather than doing:

```
if (await this.checkHyperBeamPromise) ...
```
we can do:
```
if (await this.checkHyperBeamCompatibility())
```

Which itself can check this.checkHyperBeamPromise internally and also handle retries. If this throws for some reason or has a bad status, this will allow us to revalidate the compatibility easily with little changes in the future.

Reply:
We should also add some error handling I think here, along with logging of the status / errors

Reply:
Agreed, we don't need to add retries now, but we can memoize _in the function_ instead of checking the promise, this way if we do eventually change the pattern we can simply modify the logic in the function and not have to worry about changing the locations using this.promiseResult.

Its not a huge change to the current pattern:
```suggestion
  private async checkHyperBeamCompatibility(): Promise<boolean> {
    if (this.checkHyperBeamPromise !== undefined) {
      return this.checkHyperBeamPromise;
    }

    const res = await fetch(
      `https://permanode.xyz/${this.processId}~process@1.0/now/cache`,
      {
        method: 'HEAD',
      },
    );

    let isHyperBeamCompatible = false;

    if (res.ok) {
      isHyperBeamCompatible = true;
    }

    this.checkHyperBeamPromise = Promise.resolve(isHyperBeamCompatible);

    return isHyperBeamCompatible;
  
```

Reply:
constructor call would remain, and then using it would look like:
```ts
await this.checkHyperBeamCompatibility()
```

and it would use the memoized value

Reply:
yeah - something like that, just confirming the memoize is called and hyperbeam used. we can add with some more testing.